semana4/lista-de-exercicios/index.js

```diff
@@ -0,0 +1,406 @@
+
+/* 1. Cria uma função 'conversorDeMoeda' que recebe um parâmetro (valorEmDolar). Dentro da função a variável 
+      'cotacao' recebe um Number pelo prompt informando a cotação do dólar, e retorna o valor passado no parâmetro
+      multiplicado pela variável 'cotacao', que armazenou o valor informado no prompt.
+      A variável 'meuDinheiro' recebe a função 'conversorDeMoeda' com parâmetro 100.
+      'meuDinheiro' é exibida no console. Caso valor informado no prompt seja 6, o valor exibido no console será 600.
+      
+    2. Cria uma função 'investeDinheiro' com dois parâmetros (tipoDeInvestimento, valor). Dentro da função contém 
+       uma variável 'valorAposInvestimento' e um 'Switch' que analisa a string recebida no parâmetro 'tipoDeInvestimento'.
+       Caso (case) tipoDeInvestimento seja "Poupança" a variável valorAposInvestimento será igual o 'valor' passado como 
+       parâmetro multiplicado por 1.03, assim sendo com as outras opções: (case "Renda fixa" * 1.05), ("CDB" * 1.06), 
+       ("Ações" * 1.1), a opção default será caso tipoDeInvestimento não esteja dentre essas opções (será mostrado um alert)
+       A função retorna 'valorAposInvestimento'.
+       A variável novoMontante recebe a função com os parâmetros ("Ações", 150)
+       A variável segundoMontante recebe a função com os parâmetros("Tesouro Direto", 200)
+       É impresso no console 'novoMontante' com valor 165. 'segundoMontante' exibe o alert
+
+    3. Cria o array 'numeros' com valores e os arrays 'array1' e 'array2' vazios.
+       Com o 'for of' para iterar no array 'numeros', é feito uma condição com 'if'; se o resto da divisão por 2 for 0,
+       este valor é adicionado no array1 (array1.push(numero)), Se for diferente de 0, o valor é adicionado no array2 (array2.push(numero))
+       No console é impresso a quantidade total de números no array 'numeros' (14), e os valores contidos nos array1 (6) e array2 (8).
+       
+    4. Cria um array 'numeros' duas variáveis, 'numero1' com valor 'Infinity' e 'numero2' com valor 0.
+       Dentro de um 'for of' testa-se duas condições.
+       1ª ------ 
+       2ª Para cada iteração no array numeros, o valor for maior que a variável numero2, a mesma recebe o valor do array. Para esse teste
+       é exibo no console o valor 1590.
+
+*/
+
+
+
+
+
+// Exercícios de Lógica de Programação
+
+// 1. for - for of - forEach 
+
+/*const numeros = [3,4,5,6,7];
+let soma = 0;
+
+for (numero of numeros) {
+    soma += numero;
+}
+
+console.log(soma); */       
+
+
+// 2. 
+
+/* a) false
+   b) false
+   c) true
+   d) true
+   e) true     
+*/
+
+
+// 3. Este código não funciona, pois ele entra em um loop infinito porque a variável 'i'
+//    sempre vai ser menor que 'quantidadeDeNumerosPares' 
+
+// Código alterado para funcionar
+
+/* const quantidadeDeNumerosPares = 3;
+    let i = 0;
+    while(i <= (quantidadeDeNumerosPares + 1)) {
+        console.log(i);
+        i += 2;
+        quantidadeDeNumerosPares -1;
+    } 
```

Comment:
O gabarito certinho com uma sugestão de implementação: 

Você tem que escrever um código que, dado um número N, ele imprima (no console) os N primeiros números pares 
(por exemplo, se N for 3, você deve imprimir 0, 2 e 4).  Um colega seu disse que já começou esta tarefa, mas não conseguiu terminar. Dê uma olhada no código dele:
  const quantidadeDeNumerosPares
let i = 0
while(i <= quantidadeDeNumerosPares) {
  console.log(i*2)
}

Este código funciona? Por quê? Caso não funcione, corrija a implementação dele.RESPOSTA:

Não. Há dois erros principais:
  1) O Loop nunca vai para de rodar.Perceba que a condicação do loop é quando a variável i chegar no valor
da variável quantidadeDeNumerosPares. Mesmo ela sendo declarada como "let", não
há nenhuma linha de código que atualize o valor desta variável. Desta forma, o loop nunca vai parar de rodar.
2) A condição de parada está errada.Quando usamos while, o código dentro dele irá rodar até que a condição dentro
dele se assuma o valor "false". Do jeito que está feito ele vai rodar
uma vez a mais do que precisa.Vamos considerar que quantidadeDeNumerosPares seja 3
O loop ira imprimir no console

// para i = 0
Console: 0

// para i = 1
Console: 2

// para i = 2
Console: 4

// para i = 3 (o código roda, porque i <= quantidadeDeNumerosPares é "true". i é igual a 3)
Console: 6

// para i = 4
// o loop vai parar de rodar porqu i <= quantidaddeDe NUmerosPares é "false". i é maior do que 3)

Uma possível correção do código está abaixo


function ex3NumerosPares() {
  const quantidadeDeNumerosPares
  let i = 0
  while (i < quantidadeDeNumerosPares) {
    console.log(i * 2)
    i++
  }
}